File: Quorum3Plugins/SystemPlugin/src/plugins/quorum/Libraries/System/Properties.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package plugins.quorum.Libraries.System;

/**
 *
 * @author astefik
 */
public class Properties {
    public java.lang.Object me_ = null;
    
    public void SetPropertyNative(String key, String value) {
        System.setProperty(key, value);
    }
    
    public String GetPropertyNative(String key) {
        String property = System.getProperty(key);
        return property;
    }
    public boolean HasPropertyNative(String key) {
        String property = System.getProperty(key);
        if (property == null) {
            return false;
        } else {
            return true;
        }
    }
    
    public String GetEnvironmentVariableNative(String key) {
        String env = System.getenv(key);
        return env;
    }
    
    public boolean HasEnvironmentVariableNative(String key) {
        String property = System.getenv(key);
        if (property == null) {
            return false;
        } else {
            return true;
        }
    }
}

File: Quorum3/Library/Standard/Plugins/javascript/Libraries/System/Properties.js
function plugins_quorum_Libraries_System_Properties_() {
    this.GetRunLocation = function () {
        return null;
    };
    
    this.SetPropertyNative$quorum_text$quorum_text = function (key, value) {
    };  
    
    this.GetPropertyNative$quorum_text = function (key) {
        if(key == "os.name") {
            return "Web Browser";
        }
    };  
    
    this.HasPropertyNative$quorum_text = function (key) {
        if(key == "os.name") {
            return true;
        }
    };
    
    this.GetEnvironmentVariableNative$quorum_text = function (key) {
        return "";
    };  
    
    this.GetEnvironmentVariableNative$quorum_text = function (key) {
        return false;
    };
}